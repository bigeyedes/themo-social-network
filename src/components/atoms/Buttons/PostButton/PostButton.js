import styled from 'styled-components';
import send from './send.svg';

const PostButton = styled.button`
	background: #3975F5;
	color: white;
	border: none;
	border-radius:25px;
	width: 50px;
	height: 50px;
	transition: .3s ease;
	background-image: url(${send});
	background-position: center;
	background-size: 20px auto;
	background-repeat: no-repeat;
	cursor: pointer;
	&:hover {
		opacity: 0.7
	}
`

export default PostButton;