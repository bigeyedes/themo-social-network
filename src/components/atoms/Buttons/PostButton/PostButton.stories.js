import React from 'react'
import { storiesOf } from '@storybook/react';
import PostButton from './PostButton';

storiesOf('Atoms/Buttons', module)
	.add('Send Button', () => <PostButton></PostButton>)
