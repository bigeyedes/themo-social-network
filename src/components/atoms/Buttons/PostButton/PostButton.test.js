import React from 'react'
import renderer from 'react-test-renderer';
import 'jest-styled-components'
import PostButton from './PostButton'
import sendIcon from './send.svg'

describe("Render send button component", () => {
	test("Matches the snapshot", () => {
		const tree = renderer.create(<PostButton />).toJSON()
		expect(tree).toMatchSnapshot() 
	});

	test("If is 0.7 opacity when hover", () => {
		const tree = renderer.create(<PostButton />).toJSON();
		expect(tree).toHaveStyleRule('opacity', '0.7', {
			modifier: ':hover',
		})
	});

	test("If have standard background image", () => {
		const tree = renderer.create(<PostButton />).toJSON();
		expect(tree).toHaveStyleRule('background-image', `url(${sendIcon})`)
	});

});