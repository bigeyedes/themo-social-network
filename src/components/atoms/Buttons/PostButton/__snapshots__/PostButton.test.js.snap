// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render send button component Matches the snapshot 1`] = `
.c0 {
  background: #3975F5;
  color: white;
  border: none;
  border-radius: 25px;
  width: 50px;
  height: 50px;
  -webkit-transition: .3s ease;
  transition: .3s ease;
  background-image: url(send.svg);
  background-position: center;
  background-size: 20px auto;
  background-repeat: no-repeat;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.7;
}

<button
  className="c0"
/>
`;
